File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/service/dtos/CreateServiceRequest.java
package com.slotify.backend.spring.service.dtos;

import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class CreateServiceRequest {

    @NotBlank
    @Size(max = 45)
    private String serviceName;

    @Min(5)
    @Max(480)
    @NotNull
    private Integer minutesDuration;

    @Min(100)
    @Max(1000000)
    @NotNull
    private Integer priceCent;

    private String description;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/service/CreateServiceRequest.kt
package com.raj.slotify.dtos.service

import com.google.gson.annotations.SerializedName

data class CreateServiceRequest(

    @SerializedName("serviceName")
    val serviceName: String,

    @SerializedName("minutesDuration")
    val minutesDuration: Int,

    @SerializedName("priceCent")
    val priceCent: Int,

    @SerializedName("description")
    val description: String

)
